### PR TITLE
UI integration tests, dependency updates, and enable auth by default

### PR DESCRIPTION
**Pull request by rolandpoulter**

### CODE REVIEW

File: src/common/messengers/RackHDRestAPIv2_0.js [hunk 1]
```diff
@@ -18,6 +18,11 @@ if (config.check('Enable_RackHD_API_Auth')) {
     new Swagger.ApiKeyAuthorization('Authorization', 'JWT ' + TOKEN, 'header');
 }
 
+// console.log(API, TOKEN, authorizations);
```
Comment (yyscamper) [inline src/common/messengers/RackHDRestAPIv2_0.js]: Please remove those unused code.

File: test/specs/workflows-spec.js [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+          operationsCenterGraph &&
+          activeOperationsList) return done();
+      setTimeout(check, 50);
+    };
+    window.location.hash = '#/oc';
+    check();
+  });
+
+  it('is ok', function() {
+    expect(true).to.be.ok;
```
Comment (yyscamper) [inline test/specs/workflows-spec.js]: This unit-test does nothing, please remove it.

File: test/specs/dashboard-spec.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    };
+    check();
+  });
+
+  after(function(done) {
+    done();
+  });
+
+  it('is ok', function() {
+    expect(true).to.be.ok;
```
Comment (yyscamper) [inline test/specs/dashboard-spec.js]: This unit-test does nothing, please remove it.

File: test/specs/workflows-spec.js [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+          let firstWorkflowTask = document.querySelector('div[data-id="' + firstWorkflowTaskId + '"]');
+          if (firstWorkflowTask) return done();
+          setTimeout(check, 50);
+        };
+        window.location.hash = '#/oc/' + this.workflow.instanceId;
+        check();
+      });
+
+      it('is ok', function() {
+        expect(true).to.be.ok;
```
Comment (yyscamper) [inline test/specs/workflows-spec.js]: Remove this unit-test because it does nothing.

File: test/specs/workflows-spec.js (hunk 2 above)
Comment (rolandpoulter) [inline test/specs/workflows-spec.js]: Actually this kind of unit test does in fact do something. As well as all the others. They help inform the developer that the `before` stage finished without error. Giving clear confirmation is useful.

File: src/common/lib/Messenger.js [hunk 5]
```diff
@@ -33,7 +33,11 @@ export default class Messenger extends EventEmitter {
 
     this.removeAllListeners();
 
-    this.connection = new WebSocket(this.url);
+    try {
+      this.connection = new WebSocket(this.url);
+    } catch (err) {
```
Comment (rolandpoulter) [inline src/common/lib/Messenger.js]: Yep, intentional here. If the websocket service is not properly configured we still want the UI to work right?

Comment (yyscamper) [inline src/common/lib/Messenger.js]: make sense

**Pull request merged**